### PR TITLE
Annotate Clipboard module

### PR DESCRIPTION
**Pull request by satya164**

### CODE REVIEW

File: Libraries/Components/Clipboard/Clipboard.js [hunk 1]
```diff
@@ -24,7 +25,7 @@ module.exports = {
    * }
    * ```
    */
-  getString() {
+  getString(): Promise {
```
Comment (dmmiller) [inline Libraries/Components/Clipboard/Clipboard.js]: Can you type the promise with string?
Promise<string>?

**Comment by satya164** [comment]
@facebook-github-bot shipit

**Comment by dmmiller** [comment]
I'm not sure this can be annotated while there is still a path that returns undefined (which it does in the legacy callback case)

**Comment by satya164** [comment]
@dmmiller I'll add a decallbackify function as @bestander suggested, and it should fix this. Haven't got around to it yet :)

```js
// @deprecated, will be removed on March 1 2016
// helper for deprecated callbacks pattern
function decallbackify(promise: Promise, callbacks: Array<Function>, type, warning) {
  if (callbacks.length > 0) {
    console.warn(warning);
    switch (type) {
    case 'success-first': // handles func(success, error), func(success)
      const success = callbacks[0];
      const error = callbacks[1];
      return promise.then(
        res => success(res),
        err => {
          if (error) {
            error(err);
          } else {
            throw err;
          }
        }
      );
    case 'error-first': // handles func(error, success)
      const error = callbacks[0];
      const success = callbacks[1];
      return promise.then(
        res => success(res),
        err => error(err)
      );
    case 'node': // handles func(callback)
      const callback = callbacks[0];
      return promise.then(
        res => callback(res),
        err => callback(null, err)
      );
    default:
      throw new Error('Type of callbacks not specified. Must be one of \'success-first\', \'error-first\', or \'node\'');
    }
  }

  return promise;
}
```

Then, we would do,

```js
return decallbackify(
  Platform.OS === 'android' ? RCTNetInfo.isConnectionMetered() : Promise.reject(new Error('Currently not supported on iOS')),
  arguments, 'node',
  'NetInfo.isConnectionMetered(callback) is deprecated. Use the returned Promise instead.'
);
```

**Comment by dmmiller** [comment]
Sounds good.  Will you do the decallbackify as a separate PR just so we can keep it clean?  And then you can rebase and use it.

**Comment by satya164** [comment]
@dmmiller Yup. Will do that.

**Comment by dmmiller** [comment]
Look at https://github.com/facebook/react-native/pull/5566 I like the name deprecatedPropType.  Perhaps a similar name rather than decallbackify.

**Comment by satya164** [comment]
Thanks, coming up with such cool names is my speciality (or not) :D

Will think of a better name for `decallbackify`. Please tell if you've something in mind.

**Comment by satya164** [comment]
```
callbackFromPromise
promiseToCallback
callbackWithWarning
deprecatedCallback
```

**Comment by dmmiller** [comment]
Of those, deprecatedCallback is best.

**Comment by satya164** [comment]
@dmmiller Kewl. Will send a PR tomorrow.

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Comment by satya164** [comment]
@dmmiller updated

**Comment by dmmiller** [comment]
@facebook-github-bot-0 shipit

**Comment by dmmiller** [comment]
Let's see if the bots are back in action :)

**Comment by satya164** [comment]
@dmmiller I could merge PRs yesterday, but the bot didn't close the PRs after merging.

**Comment by dmmiller** [comment]
@facebook-github-bot shipit

**Comment by mkonicek** [comment]
Commit is in master but there's a bug with commit messages not containing a link back to the PR. Normally this PR would be closed automatically. Closing manually until the bug is fixed.

**Pull request closed without merging**